http/rest-client-reactive-vanilla/src/main/java/io/quarkus/ts/http/restclient/reactive/VersionHeaderFilter.java

```diff
@@ -0,0 +1,30 @@
+package io.quarkus.ts.http.restclient.reactive;
+
+import java.io.IOException;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.ws.rs.container.ContainerRequestContext;
+import jakarta.ws.rs.container.ContainerResponseContext;
+import jakarta.ws.rs.container.ContainerResponseFilter;
+import jakarta.ws.rs.container.PreMatching;
+import jakarta.ws.rs.ext.Provider;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+/**
+ * Used for {@link RestCallerService}
+ */
+@Provider
+@PreMatching
+@ApplicationScoped
+public class VersionHeaderFilter implements ContainerResponseFilter {
+    @ConfigProperty(name = "quarkus.application.version")
+    String applicationVersion;
+
+    public VersionHeaderFilter() {
+    }
+
+    public void filter(ContainerRequestContext requestContext, ContainerResponseContext responseContext) throws IOException {
+        responseContext.getHeaders().add("Build-Version", applicationVersion);
```

Comment:
This filter is only applied when you actually call Jakarta REST resource (which you have none), so I'm not sure why you bother with `Build-Version` headers. Please at least give it context with comment or dropped dead code as IMO noone will remember why it is there in a week. Thanks

Reply:
If the point is to test that you have been and config property injection works, well, you still don't need headers that are never set.

Reply:
I've changed the filter and removed the header altering from it. Now the filter just "exists", so quarkus can inject it, but filter has no real actions. I've also added comments to it, why it is here.